proposals/3/README.md

```diff
@@ -0,0 +1,210 @@
+            *   EKS - Covered by the tekton-bot Service Account.
+    *   _storage-sa_
+        *   required
+        *   IAM roles:	
+            *   GKE - roles/iam.workloadIdentityUser, storage.objects.[create|get|delete|update]
+            *   EKS - S3Access Policy attached to an IAM/role and a Kubernetes Service Account.
+    *   _tekton-sa_
+        *   required
+        *   IAM roles:	
+            *   GKE - roles/iam.workloadIdentityUser, roles/viewer 
+            *   EKS - IAM/role and IAM Policy with general permissions, Kubernetes Service Account.
+    *   _velero-sa_
+        *   optional, only created if the user wants to use Velero
+        *   IAM roles:	
+            *   GKE - roles/iam.workloadIdentityUser, roles/storage.admin, roles/storage.objectAdmin, roles/storage.objectCreator
+            *   EKS - S3Access Policy attached to an IAM/role and a Kubernetes Service Account.
+    *   _vault-sa_
+        *   required            
+*   Storage buckets 
+    * Currently we have the following buckets
+        *   logs (required)
+        *   reports (optional, same bucket as logs if not provided)
+        *   repository (optional, used by bucketrepo if enabled)
+        *   backup (optional, used by Velero if enabled)
+    * To make things easier in the default Terraform setup we could consider using a single bucket.
+    This would simplify the conditional logic around bucket creation and would make it easier for the user to enable storage related features. 
+    This needs further investigation/discussion.
+*   Vault (optional, the user has the ability to BYO Vault instance)
+
+Permissions needed by Kubernetes service accounts for the various Cloud APIs are managed by [workload identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity).
+
+Similar to the setup in [terraform-google-jx](https://github.com/jenkins-x/terraform-google-jx), the output of applying the Terraform plan will be a `jx-requirements.yaml` which can then be used as input for `jx boot`.
+
+
+### 2.2.3 Boot and jx-requirements.yaml
+
+The plan is to leave the current Boot code, for now, untouched as much as possible.
+Boot needs to be provided with the URLs to the buckets to be used.
+These values need to be set in the existing `StorageEntryConfig.URL` fields. 
+
+For Vault, we will need to add a new field to `VaultConfig`, `VaultConfig.URL` pointing to the internal or external URL of the Vault instance to use.
+We can leave the other existing values or mark them as deprecated.
+
+`ClusterConfig` contains the already deprecated values `VaultName` and `VaultSAName`.
+It should be safe to remove them now. 
+
+
+### 2.2.4 Vault
+
+From the perspective of `jx boot`, Vault becomes a provided resource. Boot will not create a Vault instance anymore using the [Banzai Cloud Vault Operator](https://banzaicloud.com/blog/vault-operator/).
+A URL to an existing Vault instance needs to be provided via `jx-requirements.yaml`. 
```

Comment:
In this scenario, does vault exist in the same cluster? or a different cluster? or somewhere else entirely?

Reply:
If provisioned via our Terraform scripts in the same cluster, dedicated namespace. However, if the user already uses Vault and wants Jenkins X to use that instance it could live somewhere else entirely.

Reply:
In the same cluster may cause issues, you'll be sharing state and you'll introduce a dependency between terraform modules - not to mention any timing issues.